### PR TITLE
Make editor work with current version of the game

### PR DESCRIPTION
**Pull request by girtonman**
After doing some troubleshooting, it looks like the main issue that was causing it to crash was that the seed for the save was too big for a js float, so it was being rounded. 
For example:
```
  "seed": 1461819000922140955,
```
was being rounded to
```
  "seed": 1461819000922141000,
```

This would cause the game to crash when the map gen for the rounded seed would conflict with the map position info in the save file.

Adding `lossless-json` solves this issue pretty easily. I did some very simple testing and the tool seems to be working again with this fix.

### CODE REVIEW

**Comment by girtonman** [comment]
Closes the following issues:
#25 #27 #28

**Comment by zuelong** [comment]
Wow!  Thanks so much for digging and looking into this!  I'll be sure to get this deployed and maybe add some CI so I can stop doing manual deploys 😅

File: src/components/Advanced.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
+const LosslessJSON = require('lossless-json');
```
Comment (zuelong) [inline src/components/Advanced.js]: I just realized this won't actually work without proper import syntax, can you update this please?

File: src/components/Advanced.js [hunk 2]
```diff
@@ -44,6 +45,7 @@ class Advanced extends Component {
     encrypt = () => {
         try {
             const decrypted = this.state.json;
+            console.log(decrypted);
```
Comment (zuelong) [inline src/components/Advanced.js]: you'll probably want to remove this 😅

File: src/components/FileUpload.js [hunk 3]
```diff
@@ -33,7 +34,8 @@ class FileUpload extends Component {
     decrypt = (encrypted) => {
         try {
             let decrypted = this.xorWithKey(atob(encrypted), "key");
-            return JSON.parse(decrypted);
+            console.log(decrypted);
```
Comment (zuelong) [inline src/components/FileUpload.js]: You'll probably want to remove this from here as well.

**Review by zuelong — CHANGES_REQUESTED** [review]
Just a few suggestions before I can merge this in, but great work!

I'll get it tested and merged after these are corrected

File: src/components/Advanced.js (hunk 1 above)
Comment (girtonman) [reply src/components/Advanced.js]: I'm pretty new to this tech stack. I tried `import LosslessJSON from 'lossless-json';` and it was throwing null errors.

I'm open to suggestions.

File: src/components/Advanced.js (hunk 2 above)
Comment (girtonman) [reply src/components/Advanced.js]: I noticed that while doing some other work on it today. Oops!

I'll remove it.

File: src/components/FileUpload.js (hunk 3 above)
Comment (girtonman) [reply src/components/FileUpload.js]: Same as above :)

I'll remove this one as well.

**Comment by girtonman** [comment]
I've also been messing around with it and found a couple minor changes that need to be made to handle the new LosslessNumber type.

Throughout a day of testing and using it, I'm feeling even more confident in this fix.

I'm in the process of adding all of the new cards/relics to `CardsJSON` and `RelicsJSON`. Would you like me to shoe-horn them into this PR or make a new one for that? There are about 90 cards that need to be added. 😮 I haven't figured out how many relics yet though, but that should _hopefully_ be a shorter list.

File: src/components/Advanced.js (hunk 2 above)
Comment (girtonman) [reply src/components/Advanced.js]: Fixed

File: src/components/FileUpload.js (hunk 3 above)
Comment (girtonman) [reply src/components/FileUpload.js]: Fixed.

File: src/components/Advanced.js (hunk 1 above)
Comment (girtonman) [reply src/components/Advanced.js]: Read up on ES6 modules. Figured out what I needed to do based on the exports for `lossless-json`.

**Comment by girtonman** [comment]
Resolved all feedback and also made a small fix to handle the fact that the value of `upgrades` can be a number of a `LosslessValue` depending on its source (file upload or card selector),

**Comment by girtonman** [comment]
Also moved the changes I was talking about for `CardsJSON` and `RelicsJSON` to another branch that I can make a PR for once this one is wrapper up.

**Pull request closed without merging**